SensusService/Probes/User/ScriptRunner.cs

```diff
@@ -492,11 +517,11 @@ private void StartRandomTriggerCallbacksAsync()
                         DateTime triggerWindowEnd = default(DateTime);
                         DateTime now = DateTime.Now;
                         bool foundTriggerWindow = false;
-                        foreach (Tuple<int, int> randomTriggerWindow in _randomTriggerWindows)
-                            if (randomTriggerWindow.Item1 > now.Hour)
+                        foreach (Tuple<DateTime, DateTime> randomTriggerWindow in _randomTriggerWindows)
+                            if (randomTriggerWindow.Item1 > now)
```

Comment:
You're comparing the start of the trigger window to `now`. This comparison includes the date as well as the time. Initially the trigger window date will be the current date, so this will work; however, once the current day rolls over to the next day I don't think the comparison of `randomTriggerWindow.Item1 > now` will never be true. Similarly to the previous version (see left side of screen), the comparison should only be with respect to the hour and minute of `now`.